test/boost/commitlog_test.cc

```diff
@@ -1091,3 +1092,136 @@ BOOST_AUTO_TEST_CASE(test_commitlog_segment_descriptor) {
     }
 }
 
+static future<> do_test_commitlog_replay_compression(std::string_view compressor_name) {
+    auto cfg = make_shared<db::config>();
+    tmpdir tmp;
+
+    // set commitlog compression as well as a cross-restart persistent data dir
+    cfg->commitlog_compression({{ "class_name", compressor_name.data() }});
+    cfg->data_file_directories({tmp.path().string()});
+
+    co_await do_with_cql_env_thread([] (cql_test_env& env) {
+        env.execute_cql("create table t (pk text primary key, v text)").get();
+        env.execute_cql("insert into ks.t (pk, v) values ('apa', 'ko')").get();
+
+        // fake unclean shutdown by simply dropping all commitlog segments.
+        env.db().invoke_on_all([](auto& db) -> future<> {
+            co_await db.commitlog()->sync_all_segments();
+            co_await db.commitlog()->release();
+        }).get();
+
```

Comment:
Do you mean that we're leaving commitlog segments on disk? That is directly tested in the restart, since unless we did so, no data would be readable, as we delete all sstables for the keyspace

Reply:
You're right. We create another cql_test_env etc so before and after are isolated.

In fact I'm surprised the new cql_test_env reused the old one's directory. Don't they get a randomized directory?

Reply:
Normally, yes. But we set the data directory to a fixed (temp) one we create.

Reply:
aha